packages/text-input/index.js

```diff
@@ -1,42 +1,48 @@
+import classNames from 'classnames'
 import { v1 as uuidv1 } from 'uuid'
+import s from './style.module.css'
 
-function TextInput(props) {
-  //  `field` and `form` props come from Formik
-  //  https://jaredpalmer.com/formik/docs/api/field
-  const { field, form, type, label, placeholder, theme } = props
+function TextInput({
+  className,
+  field,
+  form,
+  type = 'text',
+  label,
+  placeholder,
+  theme = {
+    background: 'light',
+  },
+}) {
+  //  Note: `field` and `form` props typically come from Formik
+  //  ref: https://jaredpalmer.com/formik/docs/api/field
+  //  However, they can also be provided in non-Formik use cases
+  //  and this component will still function as expected.
   const error = form.touched[field.name] && form.errors[field.name]
   //  Label htmlFor relies on an id on the input field, which must be
   //  unique to prevent collisions between fields or forms on the same page
   const inputId = 'u' + uuidv1()
   return (
     <div
-      className="g-text-input"
-      hidden={type === 'hidden'}
-      data-theme-bg={theme.background}
+      className={classNames(s.root, className, s[`theme-${theme.background}`], {
+        [s.hidden]: type === 'hidden',
+      })}
     >
       {label && (
-        <label htmlFor={inputId}>
-          <span className="g-type-label">{label}</span>
+        <label htmlFor={inputId} className={s.label}>
+          {label}
         </label>
       )}
       <input
         id={inputId}
         type={type}
-        data-has-error={(!!error).toString()}
```

Comment:
Was this data attr just used for styling previously?

Reply:
Yep, exactly. Also this question is a great example of why we decided we should _not_ use `data-` attributes for styling 😅